src/measure_theory/probability_mass_function/monad.lean

```diff
@@ -165,6 +165,16 @@ instance : monad pmf :=
 { pure := λ A a, pure a,
   bind := λ A B pa pb, pa.bind pb }
 
+instance : is_lawful_functor pmf :=
+{ map_const_eq := λ α β, rfl,
+  id_map := λ α x, pmf.bind_pure x,
+  comp_map := λ α β γ g h x, by simp only [functor.map, bind_bind, pure_bind] }
+
+instance : is_lawful_monad pmf :=
+{ bind_pure_comp_eq_map := λ α β f x, rfl,
+  bind_map_eq_seq := λ α β f x, rfl,
+  pure_bind := λ α β, pmf.pure_bind,
+  bind_assoc := λ α β γ, pmf.bind_bind }
```

Comment:
If you put these in `constructions.lean` then you can golf the proof to
```lean
instance : is_lawful_functor pmf :=
{ map_const_eq := λ α β, rfl,
  id_map := λ α, bind_pure,
  comp_map := λ α β γ g h x, (map_comp _ _ _).symm }

instance : is_lawful_monad pmf :=
{ bind_pure_comp_eq_map := λ α β f x, rfl,
  bind_map_eq_seq := λ α β f x, rfl,
  pure_bind := λ α β, pure_bind,
  bind_assoc := λ α β γ, bind_bind }
```
Right now we're proving `comp_map` twice.